Replicator/Replicator.cc

```diff
@@ -817,6 +820,11 @@ namespace litecore { namespace repl {
             MessageIn *response = progress.reply;
 
             if (response->isError()) {
+                if (response->getError().message == "No handler for BLIP request") {
+                    auto error = C4Error::printf(LiteCoreDomain, kC4ErrorRemoteError,
+                                                 "No handler for BLIP request 'getCollections'");
+                    return onError(error);
+                }
                 return gotError(response);
```

Comment:
Have you tried to find out why gotError not working generally? It appears that gotError is kind-of used pretty standardly.

Reply:
Also how is the error turning into 26 instead of 404?

Reply:
All `gotError(MessageIn response)` does is extract the error from a response, and pass that to `onError()`. 1. When I let that error get passed to gotError, it goes to blipToC4Error() and causes "Received unknown error" to be logged a bunch of times.

You may wonder why I didn't add handling for this error into blipToC4Error(). The reason is because "No handler for BLIP request" could be for any request that we send to SG, and I don't want to silence future potential errors. Additionally, handling it here in getCollections, I can add in the context ('getCollections') to the error message to be logged.

I changed the error into 26 (kC4ErrorRemoteError) myself. 404 is the BLIP error, in Core context I believe it should be 26 (remote error).

Reply:
I worry that remote error is too vague because now any remote error is going to stop the replicator, with a context specific message about "getCollections" (as added in the stopping entry list)

Reply:
Ah you're right, I didn't realise the message string wasn't compared as well.

Reply:
So the actual domain of the error is "BLIP". We don't have an error::Domain or C4ErrorDomain for that (so `blipToC4Error()` will cause "Received unknown error from sever" to be logged). So I either need to add a new error domain to Core, or convert the error (i.e. LiteCoreDomain, kC4RemoteError, original err msg). I imagine it will be the latter (convert the error). 
Then, we need to decide whether to: stop the Replicator on any kC4RemoteError, or add in a new conditional to onError - for that specific error - and stop the Replicator in there. I don't love the latter, but I guess it would do for now.